unit_tests/schema_tests.json

```diff
@@ -0,0 +1,579 @@
+{
+  "schema": "/application/deed/schema.json",
+  "description": "validation of deed application",
+  "test_patterns": {
+    "surname": {
+      "pattern_path": "/definitions/PrivateIndividualName/properties/surname",
+      "payload_tests": [
+        {
+          "payload": "",
+          "expected": false,
+          "description": "must not be blank"
+        },
+        {
+          "payload": "bloggs",
+          "expected": true,
+          "description": "valid name"
+        },
+        {
+          "payload": "surname with",
+          "expected": false,
+          "description": "names do not have spaces in"
+        },
+        {
+          "payload": "names_can_be_reaaly_reaaly_very__very_extremely_long",
+          "expected": true,
+          "description": "names can be long"
+        }
+      ]},
+    "forename":{
+      "pattern_path": "/definitions/PrivateIndividualName/properties/forename",
+      "payload_tests": [
+        {
+          "payload": "",
+          "expected": false,
+          "description": "must not be blank"
+        },
+        {
+          "payload": "joe",
+          "expected": true,
+          "description": "valid name"
+        },
+        {
+          "payload": "first name",
+          "expected": false,
+          "description": "names do not have spaces in"
+        },
+        {
+          "payload": "names_can_be_reaaly_reaaly_very__very_extremely_long",
+          "expected": true,
+          "description": "names can be long"
+        }
+      ]
+    },
+    "dob":{
+      "pattern_path": "/definitions/PrivateIndividualName/properties/dob",
+      "payload_tests": [
+        {
+          "payload": "",
+          "expected": false,
+          "description": "must not be blank"
+        },
+        {
+          "payload": "01/02/1975",
+          "expected": true,
+          "description": "valid date"
+        },
+        {
+          "payload": "1/2/1975",
+          "expected": false,
+          "description": "invalid date"
+        }
+      ]
+    },
+    "gender":{
+      "pattern_path": "/definitions/PrivateIndividualName/properties/gender",
+      "payload_tests": [
+        {
+          "payload": "Male",
+          "expected": true,
+          "description": "check 'Male' is ok"
+        },
+        {
+          "payload": "Female",
+          "expected": true,
+          "description": "check 'Female' is ok"
+        },
+        {
+          "payload": "Not Specified",
+          "expected": true,
+          "description": "check 'Not Specified' is ok"
+        },
+        {
+          "payload": "something else",
+          "expected": false,
+          "description": "other values are not OK"
+        },
+        {
+          "payload": "",
+          "expected": false,
+          "description": "blank is not ok"
+        }
```

Comment:
Just looking at the acceptance criteria for the old story and it's saying gender optional so having it as blank would be ok?